crates/external_services/src/email/smtp.rs

```diff
@@ -0,0 +1,187 @@
+    /// SMTP server specific configs
+    pub smtp_config: SmtpServerConfig,
+}
+
+impl SmtpServer {
+    /// A helper function to create SMTP server client
+    pub fn create_client(&self) -> Result<SmtpTransport, SmtpError> {
+        let host = self.smtp_config.host.clone();
+        let port = self.smtp_config.port;
+        let timeout = Some(Duration::from_secs(self.smtp_config.timeout));
+        let credentials = self
+            .smtp_config
+            .username
+            .clone()
+            .zip(self.smtp_config.password.clone())
+            .map(|(username, password)| Credentials::new(username.to_owned(), password.to_owned()));
+        match &self.smtp_config.connection {
+            SmtpConnection::StartTls => match credentials {
+                Some(credentials) => Ok(SmtpTransport::starttls_relay(&host)
+                    .map_err(SmtpError::ConnectionFailure)?
+                    .port(port)
+                    .timeout(timeout)
+                    .credentials(credentials)
+                    .build()),
+                None => Ok(SmtpTransport::starttls_relay(&host)
+                    .map_err(SmtpError::ConnectionFailure)?
+                    .port(port)
+                    .timeout(timeout)
+                    .build()),
+            },
+            SmtpConnection::Plaintext => match credentials {
+                Some(credentials) => Ok(SmtpTransport::builder_dangerous(&host)
+                    .port(port)
+                    .timeout(timeout)
+                    .credentials(credentials)
+                    .build()),
+                None => Ok(SmtpTransport::builder_dangerous(&host)
+                    .port(port)
+                    .timeout(timeout)
+                    .build()),
+            },
+        }
+    }
+    /// Constructs a new SMTP client
+    pub async fn create(conf: &EmailSettings, smtp_config: SmtpServerConfig) -> Self {
+        Self {
+            sender: conf.sender_email.clone(),
+            smtp_config: smtp_config.clone(),
+        }
+    }
+    /// helper function to convert email id into Mailbox
+    fn to_mail_box(email: String) -> EmailResult<Mailbox> {
+        Ok(Mailbox::new(
+            None,
+            email
+                .parse()
+                .map_err(SmtpError::EmailParsingFailed)
+                .change_context(EmailError::EmailSendingFailure)?,
+        ))
+    }
+}
+/// Struct that contains the SMTP server specific configs required
+#[derive(Debug, Clone, Default, serde::Serialize, serde::Deserialize)]
+#[serde(rename_all = "snake_case")]
+pub struct SmtpServerConfig {
+    /// hostname of the SMTP server eg: smtp.google.com
+    pub host: String,
+    /// portname of the SMTP server eg: 25
+    pub port: u16,
+    /// timeout for the SMTP server connection eg: 10
+    pub timeout: u64,
+    /// Username name of the SMTP server
+    pub username: Option<String>,
+    /// Password of the SMTP server
+    pub password: Option<String>,
```

Comment:
We can make these two fields as `Secret`, not sure if we should make `host` a `Secret` as well.